UnitySDK/Assets/ML-Agents/Scripts/Grpc/RpcCommunicator.cs

```diff
@@ -114,11 +123,16 @@ public UnityRLInitParameters Initialize(CommunicatorInitParameters initParameter
         /// <param name="brainParameters">Brain parameters needed to send to the trainer.</param>
         public void SubscribeBrain(string brainKey, BrainParameters brainParameters)
         {
-            if (m_CurrentAgents.ContainsKey(brainKey))
+            var obsSize = brainParameters.numStackedVectorObservations * brainParameters.vectorObservationSize;
```

Comment:
This isn't quite right - brainParameters.numStackedVectorObservations and vectorObservationSize determine the Agent's observation buffer size (for the purposes of Agent.AddVectorObs), but there can be other vector or visual float observations from sensors that need to be accounted for. This is computed here https://github.com/Unity-Technologies/ml-agents/blob/a8dbe2df68abe9d0a1f7daa1001178ca40cb1ab0/UnitySDK/Assets/ML-Agents/Scripts/Agent.cs#L542-L551 but the BrainParameters aren't sufficient to determine the size.

Reply:
I made the evaluation of the number of floats in the PutObservation call. Let me know what you think.